mbtiles.go

```diff
@@ -92,14 +119,48 @@ func Open(path string) (*MBtiles, error) {
 	db := &MBtiles{
 		filename:  path,
 		pool:      pool,
-		timestamp: stat.ModTime().Round(time.Second),
+		timestamp: modTime,
 		format:    format,
 		tilesize:  tilesize,
 	}
 
 	return db, nil
 }
 
+func fileModTime(path string) (time.Time, error) {
+	stat, err := os.Stat(path)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return time.Time{}, fmt.Errorf("path does not exist: %q", path)
+		}
+		return time.Time{}, err
+	}
+	return stat.ModTime().Round(time.Second), nil
+}
+
+func validateAndGetFormatAndSize(path string) (TileFormat, uint32, error) {
+	// there must not be a corresponding *-journal file (tileset is still being created)
+	if _, err := os.Stat(path + "-journal"); err == nil {
+		return 0, 0, fmt.Errorf("refusing to open mbtiles file with associated -journal file (incomplete tileset)")
+	}
+
+	// open a single connection first while we are verifying the database
+	// since there are issues closing out a connection pool on error here
+	con, err := sqlite.OpenConn(path, sqlite.SQLITE_OPEN_READONLY|sqlite.SQLITE_OPEN_NOMUTEX)
+	if con != nil {
```

Comment:
This can probably be simplified like you did above, and put in a `defer con.Close()` immediately following the `if err != nil` block.